deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/PodParentHandler.java

```diff
@@ -45,4 +49,23 @@ public DeploymentStatus buildStatusFromMetadata(V1ObjectMeta metadata, Deploymen
     }
 
     public abstract String getKind();
+    
+	public String getPodSelector(ApiClient apiClient, String selector, boolean label, String namespace) {
+		if(selector==null) {
+			return null;
+		}
+		String revision = getPodRevision(apiClient, selector, label, namespace);
+		return revision != null ? selector.concat("," + revision) : selector;
+	}
+    
+	public String getPodSelector(ApiClient apiClient,T t, String selector) {
+		if(t==null) {
+			return null;
+		}
+		String revision = getPodRevision(apiClient,t);
+		if(selector==null && revision!=null) {
+			return revision;
+		}
+		return revision != null ? selector.concat("," + revision) : selector;
+	}
```

Comment:
If t is null we should return selector
if selector is null we can return revision, no need to check if revision is null.

Reply:
but here revision may be null in that what will be the value of  selector.concat("," + revision)